api/src/test/java/com/epam/pipeline/acl/datastorage/DataStorageApiServiceCommonTest.java

```diff
@@ -0,0 +1,649 @@
+
+        assertThat(dataStorageApiService.getAvailableStoragesWithShareMount(ID)).isEmpty();
+    }
+
+    @Test
+    @WithMockUser(roles = ADMIN_ROLE)
+    public void shouldReturnAvailableStoragesForAdmin() {
+        doReturn(mutableListOf(s3bucket)).when(mockDataStorageManager).getDataStorages();
+
+        assertThat(dataStorageApiService.getAvailableStorages()).hasSize(1).contains(s3bucket);
+    }
+
+    @Test
+    @WithMockUser(username = SIMPLE_USER)
+    public void shouldReturnAvailableStoragesWhenReadPermissionIsGranted() {
+        initAclEntity(s3bucket, AclPermission.READ);
+        doReturn(mutableListOf(s3bucket)).when(mockDataStorageManager).getDataStorages();
+
+        final List<AbstractDataStorage> returnedDataStorages = dataStorageApiService.getAvailableStorages();
+
+        assertThat(returnedDataStorages).hasSize(1).contains(s3bucket);
+        assertThat(returnedDataStorages.get(0).getMask()).isEqualTo(READ_PERMISSION);
+    }
+
+    @Test
+    @WithMockUser(username = SIMPLE_USER)
+    public void shouldReturnAvailableStoragesWhenWritePermissionIsGranted() {
+        initAclEntity(s3bucket, AclPermission.WRITE);
+        doReturn(mutableListOf(s3bucket)).when(mockDataStorageManager).getDataStorages();
+
+        final List<AbstractDataStorage> returnedDataStorages = dataStorageApiService.getAvailableStorages();
+
+        assertThat(returnedDataStorages).hasSize(1).contains(s3bucket);
+        assertThat(returnedDataStorages.get(0).getMask()).isEqualTo(WRITE_PERMISSION);
+    }
+
+    @Test
+    @WithMockUser(username = SIMPLE_USER)
+    public void shouldReturnAvailableStoragesWhichPermissionIsGranted() {
+        initAclEntity(s3bucket, AclPermission.READ);
+        initAclEntity(anotherS3bucket);
+        doReturn(mutableListOf(s3bucket, anotherS3bucket)).when(mockDataStorageManager).getDataStorages();
+
+        final List<AbstractDataStorage> returnedDataStorages = dataStorageApiService.getAvailableStorages();
+
+        assertThat(returnedDataStorages).hasSize(1).contains(s3bucket);
+        assertThat(returnedDataStorages.get(0).getMask()).isEqualTo(READ_PERMISSION);
+    }
+
+    @Test
+    @WithMockUser(username = SIMPLE_USER)
+    public void shouldReturnEmptyAvailableStoragesWhenWritePermissionIsNotGranted() {
+        initAclEntity(s3bucket);
+        doReturn(mutableListOf(s3bucket)).when(mockDataStorageManager).getDataStorages();
+
+        assertThat(dataStorageApiService.getAvailableStorages()).isEmpty();
+    }
+
+    @Test
+    @WithMockUser(roles = ADMIN_ROLE)
+    public void shouldLoadDataStorageForAdmin() {
+        doReturn(s3bucket).when(mockDataStorageManager).load(ID);
+        mockUserContext(context);
+
+        assertThat(dataStorageApiService.load(ID)).isEqualTo(s3bucket);
+    }
+
+    @Test
+    @WithMockUser(username = SIMPLE_USER)
+    public void shouldLoadDataStorageWhenPermissionIsGranted() {
```

Comment:
Looks a bit confusing:
-  `@WithMockUser(username = SIMPLE_USER)` - runs test with `SIMPLE_USER`
-  `initAclEntity(s3bucket, AclPermission.READ);`- grants permission for `SIMPLE_USER`
-  `initUserAndEntityMocks(OWNER_USER, s3bucket, context);` - mocks auth for `OWNER_USER`

To my mind we should use one user - `SIMPLE_USER` for all actions.